### PR TITLE
[dif] Implement function dif_csrng_stop

### PR DESCRIPTION
**Pull request by doreis-lowrisc**
This PR only contains:

- Implementation of the `dif_csrng_stop` function
- Application of the `dif_csrng_stop` on the `sca` module

### CODE REVIEW

File: hw/top_englishbreakfast/util/sw_sources.patch [hunk 1]
```diff
@@ -5,7 +5,7 @@ index 99485c5ff..1c8e29d86 100644
 @@ -81,19 +81,6 @@ _reset_start:
  _start:
    .globl _start
-
+
```
Comment (mundaym) [inline hw/top_englishbreakfast/util/sw_sources.patch:8]: There seem to be a bunch of extra spaces being inserted. I guess you are generating the patch in a slightly different way maybe? Might be worth trying to get rid of them to make the diff smaller.

File: hw/top_englishbreakfast/util/sw_sources.patch [hunk 2]
```diff
@@ -22,6 +22,45 @@ index 99485c5ff..1c8e29d86 100644
    // Request memory scrambling and init
    li a0, TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR
    li a1, (1<<SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT)|(1<<SRAM_CTRL_CTRL_INIT_BIT)
+diff --git a/sw/device/lib/dif/dif_csrng.c b/sw/device/lib/dif/dif_csrng.c
```
Comment (mundaym) [inline hw/top_englishbreakfast/util/sw_sources.patch:25]: I don't think you should need to make these changes to dif_csrng{.c,.h}, right? The new DIF function doesn't seem to use anything earl grey specific.

Comment (alphan) [reply hw/top_englishbreakfast/util/sw_sources.patch:25]: I think so, but `sw_sources.patch` already deletes several usages of `TOP_EARLGREY_CSRNG_BASE_ADDR`. I'm not sure if those deletions are actually required or if they are there to simplify the patch. @vogelpi would you mind taking a look?

**Review by alphan — COMMENTED** [review]
SCA and DIF changes LGTM. @vogelpi would you mind taking a look at the changes in `sw_sources.patch`? Thanks!

File: hw/top_englishbreakfast/util/sw_sources.patch (hunk 1 above)
Comment (doreis-lowrisc) [reply hw/top_englishbreakfast/util/sw_sources.patch]: Spaces removed.

File: hw/top_englishbreakfast/util/sw_sources.patch (hunk 2 above)
Comment (doreis-lowrisc) [reply hw/top_englishbreakfast/util/sw_sources.patch]: I also agree that there's no must for chenging the files dif_csrng{.c,.h} as soon the English breakfast doesn't use it.
So I changed the patch to let then untouched.

Comment (doreis-lowrisc) [reply hw/top_englishbreakfast/util/sw_sources.patch]: I also agree that there's no must for chenging the files dif_csrng{.c,.h} as soon the English breakfast doesn't use it.
So I changed the patch to let then untouched.

**Review by vogelpi — APPROVED** [review]
Sorry for keeping you waiting. I was out of office and have quite some backlog. Anyway, as you already noted, the patch file shouldn't patch the CSRNG DIF, but just files under SCA where the DIF is referenced. This looks good from my side! I am happy to merge this if someone else from the software team also approves the PR.

Thanks for the PR @doreis-lowrisc !

**Comment by mundaym** [comment]
@doreis-lowrisc Can you rebase this onto the latest master? I'd like the CI tests to run again and then I'll merge it. Thanks!

**Comment by doreis-lowrisc** [comment]
> @doreis-lowrisc Can you rebase this onto the latest master? I'd like the CI tests to run again and then I'll merge it. Thanks!

Done!

**Pull request merged**